include/swift/AST/DiagnosticEngine.h

```diff
@@ -390,6 +390,17 @@ namespace swift {
       return limitBehaviorUntilSwiftVersion(limit, languageMode);
     }
 
+    /// Limit the diagnostic behavior to warning until the next future
+    /// language mode.
+    ///
+    /// This should be preferred over passing the next major version to
+    /// `warnUntilSwiftVersion` to make it easier to find and update clients
+    /// when a new language mode is introduced.
+    ///
+    /// This helps stage in fixes for stricter diagnostics as warnings
+    /// until the next major language version.
+    InFlightDiagnostic &warnUntilFutureSwiftVersion();
```

Comment:
Could we name this `warnUntilFutureSwiftMode`, `warnUntilFutureSwiftLangMode`, or `warnUntilFutureSwiftLanguageMode`? I really think we should rename the existing ones too.

Reply:
Yeah I don't really feel too strongly about the name, I think we ought to be consistent though, so I think this would probably best be done in a separate PR that renames everything

Reply:
Yes please to renaming "version" to "language mode" in all of these APIs in a follow up!